pinot-core/src/main/java/org/apache/pinot/core/periodictask/BasePeriodicTask.java

```diff
@@ -18,20 +18,39 @@
  */
 package org.apache.pinot.core.periodictask;
 
+import javax.annotation.concurrent.ThreadSafe;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
 /**
  * A base class to implement periodic task interface.
  */
+@ThreadSafe
 public abstract class BasePeriodicTask implements PeriodicTask {
+  private static final Logger LOGGER = LoggerFactory.getLogger(BasePeriodicTask.class);
+
+  // Wait for at most 5 minutes while calling stop() for task to terminate
```

Comment:
comment says 5 minutes, value says 30s?

Reply:
Missed a zero. Change comment to 30s. Thanks